File: solutions/javascript/pangram/1/pangram.js
//
// This is only a SKELETON file for the 'Pangram' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const isPangram = (sentence) => {
  const letters = sentence.replaceAll(/[^a-z]/g, (char) => {
    return /[A-Z]/.test(char) ? char.toLowerCase() : "";
  })
  return new Set(letters).size == 26;
}

File: solutions/c/pangram/1/pangram.c
#include "pangram.h"

#include <stddef.h>

bool is_pangram(const char *sentence) {
    if (sentence == NULL) return false;

    int bmp = 0;
    for (int i = 0; sentence[i] != '\0'; ++i) {
        const int idx = (sentence[i] | 32) - 'a';
        if ((unsigned)(idx) >= 26) continue;
        bmp |= (1 << idx);
    }

    return bmp == ((1 << 26) - 1);
}
